src/AxelarGatewayMultisig.sol

```diff
@@ -346,6 +346,21 @@ contract AxelarGatewayMultisig is IAxelarGatewayMultisig, AxelarGateway {
         _burnToken(symbol, salt);
     }
 
+    function mintTokenAndApproveContractCall(bytes calldata params) external onlySelf {
+        (string memory symbol, address contractAddress, uint256 amount, bytes32 payloadHash) = abi.decode(
+            params,
+            (string, address, uint256, bytes32)
+        );
+
+        _mintTokenAndApproveContractCall(symbol, amount, contractAddress, payloadHash);
+    }
+
+    function approveContractCall(bytes calldata params) external onlySelf {
+        (address contractAddress, bytes32 payloadHash) = abi.decode(params, (address, bytes32));
+
+        _mintTokenAndApproveContractCall('', 0, contractAddress, payloadHash);
```

Comment:
This kind of "overloading" of functions is a programming anti-pattern that should be avoided. No gas is being saved and an `if` has to happen soon or later. Better to just have 2 internal functions instead of one that does 2 different things depending on how it is called, and then having one caller pass in dummy data or zeros for some arguments.